agent_gateway/gateway/gateway.py

```diff
@@ -322,28 +329,49 @@ def __call__(self, input: str):
             input (str): user's natural language request
         """
         result = []
-        thread = threading.Thread(target=self.run_async, args=(input, result))
+        error = []
+        thread = threading.Thread(target=self.run_async, args=(input, result, error))
         thread.start()
         thread.join()
-        try:
-            return result[0]["output"]
-        except IndexError:
-            raise AgentGatewayError(
-                message="Unable to retrieve response. Please check each of your Cortex tools and ensure all connections are valid."
-            )
+
+        if error:
+            raise error[0]
```

Comment:
This exception handling is a much cleaner output from the traces we were seeing being raised in the individual threads. At some point we may want to explore delivering all of the raised errors and investigating additional Exception types that we aren't capturing for today (meaning other than `asyncio.CancelledError` and `RuntimeError`).